### ISSUE TITLE
Thanos with duplicate results more than 5 minutes

### ISSUE THREAD

**Issue by antikilahdjs** [bitnami/charts#18422]
### Name and Version

bitnami/thanos:v0.31.0

### What architecture are you using?

amd64

### What steps will reproduce the bug?

1.Install the helm using the normal way
2.Configure S3 miniO as bucket
3.I have the Prometheus CRD and I configured the Thanos Sidecar
4.Everything works perfectly and the remote_write has been configured to sent all data to S3

### Are you using any custom parameters or values?
```yaml
        query:
        ## @param query.enabled Set to true to enable Thanos Query component
        ##
        enabled: true
        ## @param query.logLevel Thanos Query log level
        ##
        logLevel: info
        ## @param query.logFormat Thanos Query log format
        ##
        logFormat: logfmt
        ## @param query.replicaLabel Replica indicator(s) along which data is de-duplicated
        ##
        replicaLabel: [prometheus_replica]
        ## Dynamically configure store APIs using DNS discovery
        ## @param query.dnsDiscovery.enabled Enable store APIs discovery via DNS
        ## @param query.dnsDiscovery.sidecarsService Sidecars service name to discover them using DNS discovery
        ## @param query.dnsDiscovery.sidecarsNamespace Sidecars namespace to discover them using DNS discovery
        ##
        dnsDiscovery:
          enabled: true
          sidecarsService: "prometheus-k8s"
          sidecarsNamespace: "kubesphere-monitoring-system"
```
### What is the expected behavior?

If I search a query before 5 minutes I got the correct result (dedup)

![image](https://github.com/bitnami/charts/assets/33559260/57a3ac28-994b-441f-b91f-7249367cc18a)

### What do you see instead?

I have 2 Prometheus as operator replicas with Thanos then send data to my storage like miniO.
 
 If searching some query for more than 1 or 2 hours the result is twice duplicated like below
 
 
![image](https://github.com/bitnami/charts/assets/33559260/a5d3ed90-4be3-4b38-ac4b-f8577f2b1ae8)

It brings many results and the dudup is not working or I did something wrong.

### Additional information

I have the Thanos sidecar enabled in my Prometheus CRD and RemoteWrite to Storegateway.

Please help me to identify it or if it is some bug in the version that I am using.

My Prometheus wasn't configured with the external labe then the default label is prometheus_replica. I don't know if I put some unnecessary values into the helm

![image](https://github.com/bitnami/charts/assets/33559260/f6fd90cd-f1c0-44e8-9707-ad83fbe41a6a)

Thanks

**Comment by carrodher** [comment]
It appears that the issue may not be directly related to the Bitnami Thanos container image or Helm chart, but rather to how the application is being utilized or configured in your specific environment.

This seems to be a unique use case that might be challenging to replicate on our end, as it appears closely tied to your particular scenario.

For inquiries about the application itself, customizing its content, or questions about technology and infrastructure usage, we highly recommend referring to the forums and user guides provided by the project responsible for the application or technology.

With that said, we'll keep this ticket open until the stale bot automatically closes it, in case someone from the community contributes valuable insights.

**Comment by antikilahdjs** [comment]
@carrodher thank you so much and the issue was the thanos version in 0.31 now the Team worked and released the 0.32 rc main and was fixed. I believe in some days you guys from Bitnami will update it too. Thank you
